controller/adoptopenebs/admissionserver.go

```diff
@@ -0,0 +1,60 @@
+package adoptopenebs
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"mayadata.io/openebs-upgrade/types"
+	"mayadata.io/openebs-upgrade/unstruct"
+	"mayadata.io/openebs-upgrade/util"
+)
+
+// formAdmissionServerConfig forms the desired OpenEBS CR config for admission-server.
+func (p *Planner) formAdmissionServerConfig(admissionServer *unstructured.Unstructured) error {
```

Comment:
isn't adopt only about setting labels to existing resources?
What was the need to have logic that does so many stuff

Reply:
Adoption forms the OpenEBS CR based on the OpenEBS configuration present in the user's cluster.